R/read_from_dhis2-helpers.R

```diff
@@ -1,56 +1,72 @@
+  checkmate::assert_character(base_url,
+                              len = 1L, any.missing = FALSE,
+                              null.ok = FALSE)
   url <- file.path(base_url, "api", "me")
   resp <- httr::GET(url, httr::authenticate(username, password))
   httr::stop_for_status(resp)
-  R.utils::cat("\nLogged in successfully!")
+  message("\nLogged in successfully!")
 }
 
 #' Subset fields when reading from DHIS2
 #'
 #' @param fields vector of fields to select from the data frame
 #' @param data the input data frame
 #'
-#' @return the data frame with the fields of interest
+#' @return an object of type `data.frame` with the data that contains only the
+#'    fields of interest
 #' @examples
+#' \dontrun{
 #' results <- dhis2_subset_fields(
-#'  fields = c("dataElement","period","value"),
-#'  data = readepi(
-#'   credentials_file = system.file("extdata", "test.ini", package = "readepi"),
-#'    project_id = "DHIS2_DEMO",
-#'    dataset = "pBOMPrpg1QX,BfMAe6Itzgt",
-#'    organisation_unit = "DiszpKrYNg8",
-#'    data_element_group = NULL,
-#'    start_date = "2014",
-#'    end_date = "2023",
-#'    fields = c("dataElement","period","value")
-#'  )$data
-#')
-#' @export
-dhis2_subset_fields <- function(fields, data) {
+#'   data = readepi(
+#'     credentials_file   = system.file("extdata", "test.ini",
+#'     package = "readepi"),
+#'     data_source        = "https://play.dhis2.org/dev",
+#'     dataset            = "pBOMPrpg1QX,BfMAe6Itzgt",
+#'     organisation_unit  = "DiszpKrYNg8",
+#'     data_element_group = NULL,
+#'     start_date         = "2014",
+#'     end_date           = "2023",
+#'     fields             = c("dataElement", "period", "value")
+#'   )$data,
+#'   fields               = c("dataElement", "period", "value")
+#' )
+#' }
+#' @keywords internal
+dhis2_subset_fields <- function(data, fields) {
+  checkmate::assert_data_frame(data,
+                               min.rows = 1L, null.ok = FALSE,
+                               min.cols = 1L)
   checkmate::assert_vector(fields,
-                           any.missing = FALSE, min.len = 1,
-                           null.ok = TRUE, unique = TRUE
-  )
-  checkmate::assert_data_frame(data, null.ok = FALSE)
+                           min.len = 1L, null.ok = TRUE,
+                           any.missing = FALSE)
   if (!is.null(fields)) {
-    if (is.character(fields)) fields <- unlist(strsplit(fields, ",",
-                                                        fixed = TRUE))
+    if (is.character(fields)) {
+      fields <- unlist(strsplit(fields, ",",
+                                fixed = TRUE))
+    }
     idx <- which(fields %in% names(data))
-    if (length(idx) == 0) stop("Specified column not not!\nThe data contains the
-                               following column:\n", names(data))
+    if (length(idx) == 0L) stop(sprintf("Specified column not found!
+    The data contains the following columns:
+    %s", names(data)))
```

Comment:
```suggestion
    if (length(idx) == 0L) {
      stop(
        "Specified column not found!",
        "The data contains the following columns:",
        names(data)
      )
```

sprintf() is not necessary here.